pythonlib/src/mill/pythonlib/PythonModule.scala

```diff
@@ -25,55 +44,72 @@ trait PythonModule extends Module {
     Task.traverse(moduleDeps)(_.typeCheck)()
 
     os.call(
-      (pythonExe().path, "-m", "mypy", "--strict", sources().path),
+      (pythonExe().path, "-m", "mypy", "--strict", sources().map(_.path)),
       stdout = os.Inherit,
       cwd = T.workspace
     )
   }
 
-  def gatherScripts(upstream: Seq[(PathRef, PythonModule)]) = {
-    for ((sourcesFolder, mod) <- upstream) {
-      val destinationPath =
-        os.pwd / mod.millSourcePath.subRelativeTo(mill.api.WorkspaceRoot.workspaceRoot)
-      os.copy.over(sourcesFolder.path / os.up, destinationPath)
-    }
-  }
-
   def run(args: mill.define.Args) = Task.Command {
-    gatherScripts(Task.traverse(moduleDeps)(_.sources)().zip(moduleDeps))
-
     os.call(
-      (pythonExe().path, sources().path / mainFileName(), args.value),
-      env = Map("PYTHONPATH" -> Task.dest.toString),
-      stdout = os.Inherit
+      (pythonExe().path, script().path, args.value),
+      env = Map(
+        "PYTHONPATH" -> transitiveSources().map(_.path).mkString(":"),
+        "PYTHONPYCACHEPREFIX" -> (T.dest / "cache").toString
+      ),
+      stdout = os.Inherit,
+      cwd = T.dest
     )
   }
 
+  override def defaultCommandName(): String = "run"
+
+  /**
+   * Opens up a Python console with your module and all dependencies present,
+   * for you to test and operate your code interactively.
+   */
+  def console(): Command[Unit] = Task.Command(exclusive = true) {
```

Comment:
we should use repl i guess coz Li Hayoi said this in my PR

Reply:
hmm, well there's also a "console" task https://github.com/com-lihaoyi/mill/blob/5e3b6bf1c4a8c2c62bbaeb9398c311f252ee3f71/scalalib/src/mill/scalalib/ScalaModule.scala#L431.

The difference is that `repl` uses the ammonite interpreter, and `console` the basic scala interpreter. If we're trying to keep similarity, I would suggest to use `console`, since it's also the more bare-bones solution. Maybe we could have `repl` use something like `ipython`?

Reply:
ya we can create two one is repl(ipython) and other is console

it's great !!!